ceph-releases/luminous/ubuntu/16.04/daemon/clean_container.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+INITIAL_SIZE=$(du -sm / 2>/dev/null | awk {'print $1}')
+DEBUG=
+
+# Step 1
+# Let's remove easy stuff
+strip -s /usr/local/bin/{confd,forego,kubectl}
+strip -s /usr/bin/{crushtool,monmaptool,osdmaptool}
+rm -rf /usr/share/{doc,info,locale,man}/*
+rm -f /usr/bin/{etcd-tester,etcd-dump-logs}
+
+# Some logfiles are not empty, there is no need to keep them
+for logfile in $(find /var/log/ -type f); do
+    echo > $logfile
+done
+
+STEP1_SIZE=$(du -sm / 2>/dev/null | awk {'print $1}')
+REMOVED_STEP1_SIZE=$(($INITIAL_SIZE - $STEP1_SIZE))
+echo "Stripping process: Step 1 removed ${REMOVED_STEP1_SIZE}MB"
+
+# Step 2
+# Let's remove stuff than can be discussed
+
+# Let's strip the ceph libraries
+strip -s /usr/lib/ceph/erasure-code/*
+strip -s /usr/lib/rados-classes/*
+strip -s /usr/lib/python2.7/dist-packages/{rados,rbd,rgw}.x86_64-linux-gnu.so
+
+rm -f /usr/lib/ceph/mgr/dashboard/static/AdminLTE-*/plugins/datatables/extensions/TableTools/images/psd/* # Photoshop files inside a container ?
+
+# Let's remove all the pre-compiled python files, if needed they we'll be rebuilt
+find  / -xdev -name "*.pyc" -exec rm {} \; # 23MB of pyc
+
+# ceph-dencoder is only used for debugging, compressing it saves 10MB
+# If needed it will be decompressed
+gzip -9 /usr/bin/ceph-dencoder
+
+# Timezone is not configured so let's remove the zoneinfo (8MB)
+dpkg --purge --force-all tzdata
+
+# Removing perl as we don't need it
+apt-get remove -y --auto-remove perl
```

Comment:
What is pulling in `perl` to begin with? This would be great to document the reason in the comments.

Reply:
System base is pulling it

Reply:
Added a note for that. thx.

Reply:
I wonder if it's something else that pulls in "perl"?

Running `apt-get remove -y --auto-remove perl` in a vanilla `ubuntu:xenial` container says the package is not installed. But perl-base is installed, and many things like "apt" depend on it.

Reply:
Look at my comment for perl-base at line 45.